core/src/main/scala/org/apache/spark/deploy/rest/SubmitRestProtocolMessage.scala

```diff
@@ -36,7 +36,7 @@ import org.apache.spark.util.Utils
  *   (2) the Spark version of the client / server
  *   (3) an optional message
  */
-@JsonInclude(Include.NON_NULL)
+@JsonInclude(Include.NON_ABSENT)
```

Comment:
This is the core of the change to maintain behavior, as NON_NULL changed behavior in 2.7 and 2.8